server/wavelength/chat-plugins/economy.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const ECON = require('../modules/economy/index.js');
+const econWL = new ECON('stardust', 'money');
+
+const DEFAULT_LINES = 50;
+
+exports.commands = {
+	currency: 'stardust',
+	dust: 'stardust',
+	stardust: {
+	    give(target, room, user) {
+			if (!this.can('ban')) return false;
+
+			let [targetU, value] = target.split(',');
+			let targetUser = toId(targetU);
+			if (!targetUser || !value) return this.parse('/help stardust');
+
+			let amountGiven = econWL.award(targetUser, value);
+			if (!amountGiven) return this.errorReply('Indefinite amount; use whole numbers');
+
+			econWL.log(`${user.userid} gave ${targetUser} ${value} ${econWL.grammarizeName(0)}`);
+			return this.sendReply(`${targetU} has been awarded with ${amountGiven}!`);
+		},
+
+	    take(target, room, user) {
+			if (!this.can('ban')) return false;
+
+			let [targetU, value] = target.split(',');
+			let targetUser = toId(targetU);
+			if (!targetUser || !value) return this.parse('/help stardust');
+			if (econWL.get(targetU) < value) return this.errorReply(`You can't take more ${econWL.grammarizeName(0)} than they have!`);
+
+			let amountLeft = econWL.remove(targetUser, value);
+			if (!amountLeft) return this.errorReply('Indefinite amount.');
+
+			econWL.log(`${user.userid} took ${value} ${econWL.grammarizeName(0)} from ${targetUser}`);
+			return this.sendReply(`${value} ${econWL.grammarizeName(0)} has been removed from ${targetU}'s account, they now have ${amountLeft}!`);
+		},
+
+		transfer(target, room, user) {
+			let [targetU, value] = target.split(',');
+			let targetUser = toId(targetU);
+			if (!targetUser || !value) return this.parse('/help stardust');
+			if (user.userid === targetUser) return this.errorReply(`You can't transfer ${econWL.grammarizeName(0)} to yourself!`);
+			if (econWL.get(user.userid) < value) return this.errorReply(`You can't transfer more ${econWL.grammarizeName(0)} than you have!`);
+
+			let amountLeft = econWL.remove(user.userid, value);
+			let amountGiven = econWL.award(targetUser, value);
```

Comment:
It feels really weird that `econWL.{award, remove}` return the grammarised string, but then you use the same construct with `value` again in the log line below. I propose `const [transferred, amountLeft] = econWL.transfer(user.id, targetUser, value)`?

Reply:
It's the fact that both `amountLeft` and `amountGiven` both return a string using the `${value} ${econWL.grammarizeName(value)}` template, but `amountGiven`'s string is never used (it's only checked against being falsy) and then you still use the `${value} ${econWL.grammarizeName(value)}` anti-pattern latter in the log. It's just the waste of a formatted name string if you're going to manually build another one.